test/src/main/java/org/axonframework/test/matchers/Matchers.java

```diff
@@ -156,34 +156,68 @@ public static Matcher<List<CommandMessage<?>>> noCommands() {
     }
 
     /**
-     * Matches against each event of the same runtime type that has all field values equal to the fields in the
-     * expected
+     * Matches against each event of the same runtime type that has all field values equal to the fields in the expected
      * event. All fields are compared, except for the aggregate identifier and sequence number, as they are generally
      * not set on the expected event.
      *
      * @param expected The event with the expected field values
      * @param <T>      The type of event to match against
      * @return a matcher that matches based on the equality of field values
+     * @deprecated Please use the {@link #deepEquals(Object)} instead.
      */
+    @Deprecated
     public static <T> EqualFieldsMatcher<T> equalTo(T expected) {
         return new EqualFieldsMatcher<>(expected);
     }
 
     /**
-     * Matches against each event of the same runtime type that has all field values equal to the fields in the
-     * expected
+     * Matches against each event of the same runtime type that has all field values equal to the fields in the expected
      * event. All fields are compared, except for the aggregate identifier and sequence number, as they are generally
      * not set on the expected event.
      *
      * @param expected The event with the expected field values
      * @param filter   The filter describing the Fields to include in the comparison
      * @param <T>      The type of event to match against
      * @return a matcher that matches based on the equality of field values
+     * @deprecated Please use the {@link #deepEquals(Object, FieldFilter)} instead.
```

Comment:
Explain why as well. People need to know why they will need to change almost all the tests in their application

Reply:
@deprecated Please use the {@link #deepEquals(Object, FieldFilter)} instead. Using this method will lead to unwanted exceptions when running on JDK 17 and up due to Reflection changes when comparing primitives

Reply:
Also, what are our rules for deprecation? Do we want to deprecate something in a patch release? In my old projects Sonar would go haywire on deprecation warnings over this

Reply:
On point their @Morlack, will adjust the description. And, thanks for the suggestion too!